lib/Sema/CSSimplify.cpp

```diff
@@ -1135,6 +1175,28 @@ class ArgumentFailureTracker : public MatchCallArgumentListener {
   getExtraneousArguments() const {
     return ExtraArguments;
   }
+
+private:
+  bool areMissingArgsInvalid(ConstraintLocator *locator) {
```

Comment:
Maybe we should convert this into a filter over `MissingArguments` in argument tracker instead and only record the ones which come before the token?

Reply:
Or even better - ignore missing argument recorded if it comes after the token that way destructor of the `ArgumentFailureTracker` could stay untouched?